plugins/kubernetes/app/models/kubernetes/role_validator.rb

```diff
@@ -140,7 +141,7 @@ def validate_project_and_role_consistent
         kind = resource[:kind]
 
         label_paths = metadata_paths(resource).map { |p| p + [:labels] } +
-          if resource.dig(:spec, :selector, :matchLabels)
+          if resource.dig(:spec, :selector, :matchLabels) || resource[:kind] == "Deployment"
```

Comment:
aren't there a few of these like that? worth having an array of kinds, ie. job, cronjob, replicaset, etc. ?

Reply:
job: no, cronjob: no, replicaset: nobody deploys that ... so just leaves Deployment ... and that's were we saw the bug 🤷‍♂